### PR TITLE
parser: Fix for field name output in multi statements | tidb-test=pr/2070

### LINKED ISSUE #39662: Column names incorrect when using a non default delimiter
## Bug Report

This is with MySQL Shell 8.0 as client. I would expect that this also happens with MySQL Client.

```
sql> delimiter $$

sql> SELECT 1; SELECT 2; $$
+--------------+
| 1; SELECT 2; |
+--------------+
|            1 |
+--------------+
1 row in set (0.0004 sec)

+----+
| 2; |
+----+
|  2 |
+----+
1 row in set (0.0004 sec)

sql> SELECT 1 AS 'a'; SELECT 2 AS 'b'; $$
+---+
| a |
+---+
| 1 |
+---+
1 row in set (0.0004 sec)

+---+
| b |
+---+
| 2 |
+---+
1 row in set (0.0004 sec)

sql> delimiter ;

sql> SELECT tidb_version()\G
*************************** 1. row ***************************
tidb_version(): Release Version: v6.4.0
Edition: Community
Git Commit Hash: cf36a9ce2fe1039db3cf3444d51930b887df18a1
Git Branch: heads/refs/tags/v6.4.0
UTC Build Time: 2022-11-13 05:25:30
GoVersion: go1.19.2
Race Enabled: false
TiKV Min Version: 6.2.0-alpha
Check Table Before Drop: false
Store: tikv
1 row in set (0.0003 sec)
```

I expected this to have the same column names as when the default delimiter is used:
```
sql> SELECT 1; SELECT 2;
+---+
| 1 |
+---+
| 1 |
+---+
1 row in set (0.0006 sec)
+---+
| 2 |
+---+
| 2 |
+---+
1 row in set (0.0002 sec)
```

### PR DESCRIPTION
**Pull request by mjonss**
### What problem does this PR solve?

Issue Number: close #39662

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: parser/parser_test.go [hunk 1]
```diff
@@ -7142,3 +7142,21 @@ func TestTTLTableOption(t *testing.T) {
 
 	RunTest(t, table, false)
 }
+
+func TestMultiStmt(t *testing.T) {
+	p := parser.New()
+	stmts, _, err := p.Parse("SELECT 'foo'; SELECT 'bar','baz' ; select 'foo' , 'bar', 'baz';select 1", "", "")
```
Comment (dveeden) [inline parser/parser_test.go:7148]: Maybe a fieldname that contains a `;` as well?

```
mysql> SELECT 'foo;bar';
+---------+
| foo;bar |
+---------+
| foo;bar |
+---------+
1 row in set (0.00 sec)
```

**Review by dveeden — APPROVED** [review]
```
mysql> delimiter $$
mysql> SELECT 1; SELECT 2; $$
+---+
| 1 |
+---+
| 1 |
+---+
1 row in set (0.00 sec)

+---+
| 2 |
+---+
| 2 |
+---+
1 row in set (0.00 sec)

mysql> SELECT tidb_version()\G
*************************** 1. row ***************************
tidb_version(): Release Version: v6.6.0-alpha-268-gb8aa8025b
Edition: Community
Git Commit Hash: b8aa8025bf0c4cae2ecb16347e6e82d03a396898
Git Branch: field-text-39662
UTC Build Time: 2023-01-19 16:01:15
GoVersion: go1.19.5
Race Enabled: false
TiKV Min Version: 6.2.0-alpha
Check Table Before Drop: false
Store: unistore
1 row in set (0.00 sec)
```

File: parser/parser_test.go (hunk 1 above)
Comment (mjonss) [reply parser/parser_test.go]: Done :)

**Review by tangenta — APPROVED** [review]

**Pull request merged**